keanu-python/keanu/functional/biconsumer.py

```diff
@@ -0,0 +1,20 @@
+from typing import Callable, Any
+
+from py4j.java_gateway import JavaObject
+
+
+class BiConsumer:
+
+    def __init__(self, lambda_function: Callable[[JavaObject, JavaObject], None]) -> None:
+        self.lambda_function = lambda_function
+
+    def accept(self, arg1: JavaObject, arg2: JavaObject) -> None:
+        """
+        >>> c = BiConsumer(lambda x,y : print(x + y))
+        >>> c.accept("foo", "bar")
+        foobar
+        """
+        self.lambda_function(arg1, arg2)
+
+    class Java:
```

Comment:
what does `class Java` do?

Reply:
It's a clever Py4J thing that lets you use Python objects as arguments to Java methods. This Python `BiConsumer` class implements the Java `BiConsumer` interface.

Reply:
https://www.py4j.org/advanced_topics.html#implementing-java-interfaces-from-python-callback